diff --git a/Tools/gazebo_sitl_multiple_run.sh b/Tools/gazebo_sitl_multiple_run.sh
index 292272e2bd..9019b0d21c 100755
--- a/Tools/gazebo_sitl_multiple_run.sh
+++ b/Tools/gazebo_sitl_multiple_run.sh
@@ -95,6 +95,7 @@ fi
 
 echo "Starting gazebo"
 gzserver ${src_path}/Tools/sitl_gazebo/worlds/${world}.world --verbose $ros_args &
+SERVER_PID=$!
 sleep 5
 
 n=0
@@ -136,5 +137,12 @@ else
 fi
 trap "cleanup" SIGINT SIGTERM EXIT
 
-echo "Starting gazebo client"
-gzclient
+if [ "x$HEADLESS" = "x1" ]; then
+       echo "Waiting for gazebo server"
+  wait $SERVER_PID
+else
+  echo "Starting gazebo client"
+  gzclient
+fi
+
+exit 0

