diff --git a/Tools/simulation/gazebo-classic/sitl_multiple_run.sh b/Tools/simulation/gazebo-classic/sitl_multiple_run.sh
index 1d8cdbf8bc..6921187fca 100755
--- a/Tools/simulation/gazebo-classic/sitl_multiple_run.sh
+++ b/Tools/simulation/gazebo-classic/sitl_multiple_run.sh
@@ -95,6 +95,7 @@ fi
 
 echo "Starting gazebo"
 gzserver ${src_path}/Tools/simulation/gazebo-classic/sitl_gazebo-classic/worlds/${world}.world --verbose $ros_args &
+SERVER_PID=$!
 sleep 5
 
 n=0
@@ -136,5 +137,11 @@ else
 fi
 trap "cleanup" SIGINT SIGTERM EXIT
 
-echo "Starting gazebo client"
-gzclient
+if [ "x$HEADLESS" = "x1" ]; then
+  wait $SERVER_PID
+else
+  echo "Starting gazebo client"
+  gzclient
+fi
+
+exit 0
